src/storage/v2/delta.hpp

```diff
@@ -123,6 +123,30 @@ inline bool operator==(const PreviousPtr::Pointer &a, const PreviousPtr::Pointer
 
 inline bool operator!=(const PreviousPtr::Pointer &a, const PreviousPtr::Pointer &b) { return !(a == b); }
 
+struct opt_str {
+  opt_str(std::optional<std::string> const &other) : str_{other ? new_cstr(*other) : nullptr} {}
+
+  ~opt_str() {
+    if (!str_) return;
+    auto *to_del = std::exchange(str_, nullptr);
+    delete to_del;
```

Comment:
Any reason to not just `delete str_`?

Reply:
true, should be fine. I'll modify and push it up.